java/src/com/ibm/streamsx/rest/StreamingAnalyticsConnection.java

```diff
@@ -0,0 +1,117 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2017
+ */
+package com.ibm.streamsx.rest;
+
+import java.io.IOException;
+import java.util.List;
+
+import static com.ibm.streamsx.topology.context.AnalyticsServiceProperties.SERVICE_NAME;
+import static com.ibm.streamsx.topology.context.AnalyticsServiceProperties.VCAP_SERVICES;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.ibm.streamsx.rest.primitives.Instance;
+import com.ibm.streamsx.rest.primitives.Job;
+import com.ibm.streamsx.rest.primitives.Metric;
+import com.ibm.streamsx.rest.primitives.Operator;
+import com.ibm.streamsx.topology.internal.streaminganalytics.VcapServices;
+
+public class StreamingAnalyticsConnection extends StreamsConnection{
+
+    /**
+     * Basic connection to the Streaming Analytics Instance
+     *
+     * @param credentialsFile
+     *            Credentials from the Streaming Analytics File
+     * @param serviceName
+     *            Name of the service in the file above
+     * @throws IOException
+     */
+    private StreamingAnalyticsConnection(String userName, String authToken, String url) {
+      super( userName, authToken, url );
+    }
+
+    public static StreamingAnalyticsConnection createInstance( String credentialsFile, String serviceName )
+            throws IOException {
+
+        JsonObject SAcredentials = new JsonObject();
+
+        SAcredentials.addProperty(SERVICE_NAME, serviceName);
+        SAcredentials.addProperty(VCAP_SERVICES, credentialsFile);
+
+        JsonObject service = VcapServices.getVCAPService(SAcredentials);
+
+        JsonObject credential = new JsonObject();
+        credential = service.get("credentials").getAsJsonObject();
+
+        String userId = credential.get("userid").getAsString();
+        String authToken = credential.get("password").getAsString();
+        String resourcesPath = credential.get("resources_path").getAsString();
+        String sURL = credential.get("rest_url").getAsString() + resourcesPath;
+
+        String restURL = "" ;
+        StreamingAnalyticsConnection SAConn = new StreamingAnalyticsConnection( userId, authToken, restURL ) ;
```

Comment:
Java style is to use lower/camel case for variable names. 

For a typical Java developer `SAConn` looks like a class name and makes the code jarring to read.